sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/PassThroughThenCleanup.java

```diff
@@ -79,6 +82,25 @@ public void processElement(ProcessContext c) {
   }
 
   abstract static class CleanupOperation implements Serializable {
-    abstract void cleanup(PipelineOptions options) throws Exception;
+    abstract void cleanup(ContextContainer container) throws Exception;
+  }
+
+  static class ContextContainer {
+    private PCollectionView<String> view;
+    private DoFn.ProcessContext context;
+
+    public ContextContainer(
+        DoFn.ProcessContext context, PCollectionView<String> view) {
```

Comment:
This and the type of line 90 should be `DoFn<?, ?>` - that'll get rid of the cast. I'll fix myself when merging.